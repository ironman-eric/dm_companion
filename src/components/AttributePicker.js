import AttributePickerItem from './AttributePickerItem';

function AttributePicker({name, items, onOptionChange}) {
  return (
    items.map(item =>
      <AttributePickerItem name={name} onOptionChange={onOptionChange}>{item}</AttributePickerItem>
    )    
  );
}

export default AttributePicker;
