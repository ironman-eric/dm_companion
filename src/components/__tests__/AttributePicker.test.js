import React from "react";
import renderer from "react-test-renderer";
import AttributePicker from "../AttributePicker";

describe("AttributePicker", () => {
  it("Matches DOM Snapshot", () => {
    const items = ["items1","item2"];
    const domTree = renderer.create(<AttributePicker items={items} />).toJSON();
    expect(domTree).toMatchSnapshot();
  });
});
