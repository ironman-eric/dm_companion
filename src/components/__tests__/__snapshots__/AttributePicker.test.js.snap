// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttributePicker Matches DOM Snapshot 1`] = `
Array [
  <span
    className="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2"
  >
    <input
      type="radio"
      value="items1"
    />
      
    items1
  </span>,
  <span
    className="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2"
  >
    <input
      type="radio"
      value="item2"
    />
      
    item2
  </span>,
]
`;
